lib/impure/nre.nim

```diff
@@ -516,23 +516,23 @@ iterator findIter*(str: string, pattern: Regex, start = 0, endpos = int.high): R
   let unicode = uint32(getinfo[culong](pattern, pcre.INFO_OPTIONS) and
     pcre.UTF8) > 0u32
   let strlen = if endpos == int.high: str.len else: endpos+1
-
   var offset = start
   var match: Option[RegexMatch]
+  var neverMatched = true
+
   while true:
     var flags = 0
-
     if match.isSome and
        match.get.matchBounds.a > match.get.matchBounds.b:
       # 0-len match
       flags = pcre.NOTEMPTY_ATSTART
-
     match = str.matchImpl(pattern, offset, endpos, flags)
-
+    
```

Comment:
Hmm I get what you mean - you mean trailing white space after a new line right? I've just noticed - I've been goign back and forth working between linux and mac depending on where I am and just seen that VSCode on mac os seems to be inserting trailing spaces. Shouldn't because I've got it set up same as on linux. I'll recheck the settings and recommit.